pykafka/producer.py

```diff
@@ -231,14 +231,32 @@ def _setup_owned_brokers(self):
                     self, partition.leader)
         return queued_messages
 
-    def stop(self):
-        """Mark the producer as stopped"""
-        self._running = False
+    def _stop_owned_brokers(self):
         self._wait_all()
         if self._owned_brokers is not None:
             for owned_broker in self._owned_brokers.values():
                 owned_broker.stop()
 
+    def stop(self, wait=True):
+        """Mark the producer as stopped, and wait until all messages to be sent
+
+        :param wait: whether we should wait until all messages to be sent or not
+        :type wait: bool
+        """
+        def get_queue_readers():
+            return [qr for qr in threading.enumerate() if qr.name.find('pykafka.OwnedBroker.queue_reader') == 0]
+
+        while wait and self._running:
```

Comment:
I don't understand this logic - where is `self._running` set to `False` in this loop?